.github/actions/bazel/action.yaml

```diff
@@ -73,46 +73,39 @@ runs:
-    - name: Clean up execlogs
+    - name: Build execution_log_compact_to_csv
       shell: bash
       if: (success() || failure()) && inputs.execlogs-artifact-name != ''
+      env:
+        # Used by the bazel wrapper
+        BUILDBUDDY_LINKS: '[CI Job](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})'
       run: |
-        execlogs_json_in='${{ steps.metrics-tmpdir.outputs.dir }}/execlogs.json'
-        find '${{ steps.metrics-tmpdir.outputs.dir }}' -name 'execlog-*.json' \
-          -exec cat {} \; > "$execlogs_json_in"
+        set -euo pipefail
 
-        execlogs_csv_out='${{ steps.metrics-tmpdir.outputs.dir }}/execlogs.csv'
-        if ! [ -s "$execlogs_json_in" ]; then
-          echo "no execlogs found"
-          exit 0
-        fi
+        # Here we overwrite the PATH with our custom bazel wrapper to ensure
+        # the specified commands are run with the CI-specific options included
+        # in the wrapper.
+        export BAZEL_ACTION_OLDPATH="$PATH"
+        # github.action_path doesn't seem to work
+        PATH="$PWD/.github/actions/bazel/bin:$BAZEL_ACTION_OLDPATH"
+        hash -r
 
-        # this filters out some entries from the input JSON execlogs. Without this,
-        # jq takes 3+ minutes to parse the execlogs.
-        # Effectively removes a full line of "commandArgs", and takes care of skipping
-        # multiple lines for some multiline array values we don't care about
-        prog='
-          BEGIN { arr = 0; }
-          arr == 1 && /  }],?/ { arr = 0; next }
-          arr == 1 { next }
-          /  "commandArgs":.*/ { next }
-          /  "environmentVariables": \[\{/ { arr = 1; next; }
-          /  "inputs": \[\{/ { arr = 1; next; }
-          { print }
-          '
+        # Used by the bazel wrapper
+        export BAZEL_ACTION_METRICS_OUT='${{ steps.metrics-tmpdir.outputs.dir }}'
 
-        # Turns the JSON execlogs into a CSV with the following columns:
-        #   <label>,<output path>,<hash>
-        # only the target labels matching "whitelist_pat" are considered
-        whitelist_pat='^//'
-        time (cat "$execlogs_json_in" \
-          | awk "$prog" \
-          | jq -cMr --arg whitelist_pat "$whitelist_pat" \
-              'select(.targetLabel | test($whitelist_pat)) | .targetLabel as $targetLabel | .actualOutputs | map(. | $targetLabel+","+.path+","+.digest.hash) | .[]' \
-          >"$execlogs_csv_out")
+        bazel build --config=ci //bazel:execution_log_compact_to_csv
 
+    - name: Convert execlogs to CSV
+      shell: bash
+      if: (success() || failure()) && inputs.execlogs-artifact-name != ''
+      run: |
+        for execlog in $(find '${{ steps.metrics-tmpdir.outputs.dir }}' -name 'execlog-*.zst'); do
+          zstd -f -d $execlog -o execlog
+          bazel-bin/bazel/execution_log_compact_to_csv \
+            --input_execlog=execlog \
+            --whitelist_pat='^//'
```

Comment:
In a future PR it would be interesting to remove `--whitelist_pat='^//'` such that we check the reproducibility of all bazel targets including their direct and indirect dependencies.